src/internals/utils.js

```diff
@@ -159,6 +177,8 @@ export function processStyles({ styles, theme, config }) {
     } else if (typeof val === 'object' && val.value !== undefined) {
       // Handle cases where the value comes from the `theme` returned by `createStitches`
       acc[key] = val.value;
+    } else if (typeof acc[key] === 'object' && typeof val === 'object') {
```

Comment:
Could you add a comment for this case so that it's easier to understand what is happening here when I or someone else comes back to this codebase after some time 😄

Reply:
DONE